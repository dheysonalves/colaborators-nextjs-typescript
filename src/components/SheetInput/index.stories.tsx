import { ComponentMeta, ComponentStory } from '@storybook/react'
import SheetInput from '.'

export default {
	title: 'Inputs/SheetInput',
	component: SheetInput,
	args: {
		title: 'Colaboradores'
	}
} as ComponentMeta<typeof SheetInput>

export const Default: ComponentStory<typeof SheetInput> = (args) => (
	<SheetInput {...args} />
)
