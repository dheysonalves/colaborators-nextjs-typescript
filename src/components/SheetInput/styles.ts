import styled from '@emotion/styled'
import { Container } from '../utils/Flex/flex.styles'
import { Paragraph } from '../Typography/base.index'

export const Wrapper = styled(Container)`
	justify-content: space-between;
	align-items: center;
	padding: 8px 14px;
	border-radius: 8px;
	border: 2px solid #cad6d1;
`
export const NoStyleButton = styled.button`
	background-color: transparent;
	background-repeat: no-repeat;
	border: none;
	cursor: pointer;
	overflow: hidden;
	outline: none;
	padding-top: 2px;
`

export const CustomParagraph = styled(Paragraph)`
	font-size: 16px;
	font-weight: 500;
	line-height: 24px;
`
