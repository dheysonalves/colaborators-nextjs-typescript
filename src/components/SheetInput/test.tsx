import { render, screen } from '@testing-library/react'

import SheetInput from '.'

describe('<SheetInput />', () => {
	it('should render the heading', () => {
		const { container } = render(<SheetInput title="Colaboradores" />)

		expect(
			screen.getByRole('heading', { name: /Colaboradores/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
