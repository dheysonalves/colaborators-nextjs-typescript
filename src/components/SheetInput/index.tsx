import React from 'react'

import { Wrapper, NoStyleButton, CustomParagraph } from './styles'
import { BiDotsVerticalRounded } from 'react-icons/bi'

interface SheetInputProps {
	title: string
	callback?: () => unknown
}

const SheetInput = ({ title, callback }: SheetInputProps) => {
	return (
		<Wrapper>
			<CustomParagraph tag="h1">{title}</CustomParagraph>
			<NoStyleButton onClick={callback}>
				<BiDotsVerticalRounded size={24} color="#587169" />
			</NoStyleButton>
		</Wrapper>
	)
}
export default SheetInput
