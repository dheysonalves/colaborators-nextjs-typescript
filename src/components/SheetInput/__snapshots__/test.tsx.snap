// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SheetInput /> should render the heading 1`] = `
<div
  class="css-xylfo5"
>
  <h1
    class="css-1om8i5c"
  >
    Colaboradores
  </h1>
  <button
    class="css-y3q99c"
  >
    <svg
      color="#587169"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(88, 113, 105);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 12c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </button>
</div>
`
